x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/LearningToRankRescorer.java

```diff
@@ -55,6 +55,15 @@ public TopDocs rescore(TopDocs topDocs, IndexSearcher searcher, RescoreContext r
         if (ltrRescoreContext.regressionModelDefinition == null) {
             throw new IllegalStateException("local model reference is null, missing rewriteAndFetch before rescore phase?");
         }
+
+        if (rescoreContext.getWindowSize() < topDocs.scoreDocs.length) {
```

Comment:
This check could be done in `innerBuildContext` since we have all the information we need to take a decision?

Reply:
My first intent was to add it there but the SearchExecutionContext does not have the information on the sizing.
It is part of the SearchContext which is not available in the innerBuildContext.